test/NuGet.Clients.Tests/NuGet.CommandLine.Test/ExtractOptionsTests.cs

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using Xunit;
+
+namespace NuGet.CommandLine.Test
```

Comment:
What happens if we pass a NoHttpCache option to a command that does not have the NoHttpCache option? Will the code fail gracefully? Should we add a test for that scenario?

Reply:
That's a good question. It will fail gracefully : "Unknow option: -NoHttpCache". I have covered it in a new test case.